zaza/openstack/charm_tests/test_utils.py

```diff
@@ -573,6 +573,32 @@ def get_my_tests_options(self, key, default=None):
         return self.test_config.get('tests_options', {}).get(
             '.'.join(caller_path + [key]), default)
 
+    def get_applications_with_substring_in_name(self, substring):
+        """Get applications with substring in name.
+
+        :param substring: String to search for in application names
+        :type substring: str
+        :returns: List of matching applictions
+        :rtype: List
+        """
+        status = model.get_status().applications
+        applications = []
+        for application in status.keys():
+            if substring in application:
+                applications.append(application)
+        return applications
```

Comment:
This feels a tiny bit brittle?  Is it possible to instead get at the actual charm name and see if it's called "mysql-router"?